compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -1294,6 +1294,12 @@ public void visit(BLangAnnotation annotationNode) {
 
     public void visit(BLangAnnotationAttachment annAttachmentNode) {
         annAttachmentNode.expr = rewrite(annAttachmentNode.expr, env);
```

Comment:
Can take `annAttachmentNode.expr != null` before re-writing. 
```suggestion
        if(annAttachmentNode.expr != null) {
            annAttachmentNode.expr = rewrite(annAttachmentNode.expr, env);
```